Sources/Sentry/SentryUIApplication.m

```diff
@@ -12,27 +12,45 @@ - (UIApplication *)sharedApplication
     return [UIApplication performSelector:@selector(sharedApplication)];
 }
 
+- (nullable id<UIApplicationDelegate>)getApplicationDelegate:(UIApplication *)application
+{
+    return application.delegate;
+}
+
+- (NSArray *)getApplicationConnectedScenes:(UIApplication *)application
+{
+    if (@available(iOS 13.0, tvOS 13.0, *)) {
+        if (application && [application respondsToSelector:@selector(connectedScenes)]) {
+            return [application.connectedScenes allObjects];
+        }
+    }
+    return nil;
+}
+
 - (NSArray<UIWindow *> *)windows
 {
     UIApplication *app = [self sharedApplication];
-    if (app == nil)
-        return nil;
-
-    NSMutableArray *result = [NSMutableArray new];
+    NSMutableArray *result = [[NSMutableArray alloc] init];
 
     if (@available(iOS 13.0, tvOS 13.0, *)) {
-        if ([app respondsToSelector:@selector(connectedScenes)]) {
-            for (UIScene *scene in app.connectedScenes) {
+        NSArray<UIScene *> *scenes = [self getApplicationConnectedScenes:app];
+        if (scenes != nil) {
```

Comment:
`l`: Instead of returning `nil` in `getApplicationConnectedScenes`, we could return an empty array, and then we wouldn't need the check for `nil` here. The for loop would just do nothing when iterating over an empty array.

Reply:
Tbh we dont need both. To return an empty array, which is unnecessary allocation, and to check for nil.